maven-core/src/main/resources/META-INF/maven/extension.xml

```diff
@@ -187,6 +189,7 @@ under the License.
     <exportedArtifact>org.apache.maven.resolver:maven-resolver-util</exportedArtifact>
     <exportedArtifact>org.apache.maven.resolver:maven-resolver-connector-basic</exportedArtifact>
 
+    <exportedArtifact>jakarta.inject:jakarta.inject-api</exportedArtifact>
```

Comment:
don't think it should land there, ideally we shouldn't have it (we saw that javax was an issue already) but if imposed by some part of the stack (not sure which one since I thought we stayed on javax intentionally) we should just hide it from any consumer IMHO

Reply:
Or do you mean not adding it for Maven 3.x plugins ? That would be absolutely fine with me.

Reply:
The only other solution I could think about is to rewrite the whole DI injection (Sisu + Guice) because they are not pluggable at all. I'm not ready for that...

Reply:
:thinking: I don't see why:

* injections in mojo are already bridged
* abstracting injection markers (with a maven inject annotation) is easy in guice
* abstracting the injector or a lookup (like we had/have in plexus container) is quite trivial
* abstracting a scope is not crazy (wayless than what we already have)

So overall, even if I'm not sure which case(s) you want to cover, I don't see a reason to break our original hypotheis to define a clean and maven owned API and export anything jakarta related.